include/openssl/idea.h

```diff
@@ -19,52 +19,60 @@
 # include <openssl/opensslconf.h>
 
 # ifndef OPENSSL_NO_IDEA
-# ifdef  __cplusplus
+#  ifdef  __cplusplus
 extern "C" {
-# endif
+#  endif
+#  ifndef OPENSSL_NO_DEPRECATED_3_0
 
 typedef unsigned int IDEA_INT;
 
-# define IDEA_ENCRYPT    1
-# define IDEA_DECRYPT    0
+#  define IDEA_ENCRYPT    1
+#  define IDEA_DECRYPT    0
 
-# define IDEA_BLOCK      8
-# define IDEA_KEY_LENGTH 16
+#  define IDEA_BLOCK      8
+#  define IDEA_KEY_LENGTH 16
```

Comment:
Where did we get to with deprecating/not deprecating defines like this?

Reply:
Digest sizes have been kept public.  AES exposes `AES_BLOCK_SIZE` so I guess the rest of the ciphers should retain that and the key length.

Reply:
I've exposed the block size and key length regardless of the deprecation status.  The encrypt and decrypt macros don't seem use to always expose.